### PR TITLE
Implement Application/Client Credentials  Flow 

### PR DESCRIPTION
**Pull request by misi**
Fix issue #2838

### CODE REVIEW

**Comment by shockey** [comment]
ping @ponelat @bodnia

**Comment by webron** [comment]
Regardless of the review, the build is failing, so can't be merged until fixed.

**Comment by misi** [comment]
Now lint issues, and error handling has been fixed..
Please review it!

@All Many Thanks for you help and support!

File: src/core/oauth2-authorize.js [hunk 1]
```diff
@@ -34,14 +34,44 @@ export default function authorize ( auth, authActions, errActions, configs ) {
...
-  // pass action authorizeOauth2 and authentication data through window
-  // to authorize with oauth2
-  win.swaggerUIRedirectOauth2 = {
-    auth: auth,
-    state: state,
-    callback: authActions.preAuthorizeOauth2,
-    errCb: errActions.newAuthErr
-  }
+  if (flow === "application") {
+    fetch(schema.get("tokenUrl"), {
```
Comment (bodnia) [inline src/core/oauth2-authorize.js]: Please move this to actions like for password flow 
https://github.com/swagger-api/swagger-ui/blob/master/src/core/oauth2-authorize.js#L12-L15

Actions: https://github.com/swagger-api/swagger-ui/blob/master/src/core/plugins/auth/actions.js

Comment (misi) [inline src/core/oauth2-authorize.js]: It is done. It is now the form as you requested.
Please let me know your thoughts..
THX!

**Comment by bodnia** [comment]
@misi thanks for your pr

**Pull request merged**